src/taipy/core/submission/submission.py

```diff
@@ -44,19 +44,30 @@ class Submission(_Entity, _Labeled):
     def __init__(
         self,
         entity_id: str,
+        entity_type: str,
         id: Optional[str] = None,
         jobs: Optional[Union[List[Job], List[JobId]]] = None,
         creation_date: Optional[datetime] = None,
         submission_status: Optional[SubmissionStatus] = None,
         version: Optional[str] = None,
     ):
         self._entity_id = entity_id
+        self._entity_type = entity_type
         self.id = id or self.__new_id()
-        self._jobs: Union[List[Job], List[JobId], List] = jobs or []
+        self._job_ids = [job if isinstance(job, str) else job.id for job in jobs] if jobs else []
         self._creation_date = creation_date or datetime.now()
         self._submission_status = submission_status or SubmissionStatus.SUBMITTED
         self._version = version or _VersionManagerFactory._build_manager()._get_latest_version()
 
+        self.__abandoned = False
+        self.__canceled = False
+        self.__completed = False
+        self.__failed = False
+
+        self.__running_jobs: MutableSet[str] = set()
+        self.__blocked_jobs: MutableSet[str] = set()
+        self.__pending_jobs: MutableSet[str] = set()
+
```

Comment:
❓ What is the purpose of these additional `set` fields? As far as I can see, we add/remove job ids from it in `_update_submission_status` but we never read from it!
if it is just to track the statuses of the jobs, why not just using one single `Dict[JobId, Status]`?

Reply:
Oh it's just something to hold if we still have jobs running, blocked, or pending or not. As we still need access to other job statuses, if we use a dict, we will need to loop through all job statuses to do this check, while if we keep each status to an equivalent set, we can quickly identify the existence of "running jobs" or "blocked jobs", etc. in O(1) time. Does that make sense to you?

Reply:
Ok, but is it used anywhere in the code? I can't find any...
If it is not used, why do we need those sets?

Reply:
Sorry, but I have concerns about this implementation...
Let me elaborate, so you can tell me I shouldn't actually worry :-)

By keeping such "states" in the object without any db persistence will make any Taipy-core service stateful. Example: a pod running in Kubernetes. I see at least two problems now:
 - If we need to "scale up" the pod (because we need more throughput/horse power), then the state remains in on pod but the other has an inconsistent state. The behaviour will be erratic.
 - If the pod dies for some reason, then the state/data is lost forever.
So either this state is important, and thus we need to somehow persist it, either it's not useful and we should just get rid of it.

let me know if we need to discuss it with more details.

Reply:
Hmmm for now, from what I can understand, as long as orchestrator and dispatcher is on the same application, sharing the same memory pool, this shouldn't be an issue. Previously we also rely on this fact when initiating the callback from job to update the submission status:
- To run callbacks https://github.com/Avaiga/taipy-core/blob/develop/src/taipy/core/job/job.py#L31 this function relys on callbacks already store in memory
- Assigning callbacks https://github.com/Avaiga/taipy-core/blob/develop/src/taipy/core/_orchestrator/_orchestrator.py#L81

But you made a very valid point, in the scenario where we want to scale up, separating the dispatcher to a separate application or even, a separate machine, this won't work anymore. I think we should discuss this point further :D and we should also involve @jrobinAV as I think this topic would be interesting for this ticket also https://app.zenhub.com/workspaces/taipy-613ef328affde6000ff77fb3/issues/gh/avaiga/taipy-enterprise/268

Reply:
Isn't that the dispatcher and orchestrator should always on the same machine, and we only split the jobs?

Reply:
I believe we can close this topic for now.
To summarize:
- We acknowledge that there are several places that are stateful that we don't have an approach to serialize them across multiple machines yet (including job callbacks, state of the submission entity, the event notification, etc.)
- We will not these issues right now and will review them again later

Reply:
Yes, we can close the topic.